### PR TITLE
ioc container. improve performance a little

### PR DESCRIPTION
**Pull request by alchimik**

### CODE REVIEW

**Comment by alchimik** [comment]
> You could probably even remove getCurrentTarget() altogether and just use $this->currentTarget as a micro-optimization.

I'll do it

> Is there a reason to make the reflection class static rather than an instance property? 

to make it shared for all instances of the container (probably not a good idea, I'll think think about it)

>  what benchmarks did you use to determine that my is_string() and is_array() optimizations didn't have an effect?

https://github.com/kocsismate/php-di-container-benchmarks

**Comment by opulencephp** [comment]
OK.  If the effects of my `is_string()` and `is_array()` micro-optimizations were not appreciable, I might go back to the old way (array casting and looping through that array).

File: src/Opulence/Ioc/Container.php [hunk 1]
```diff
@@ -289,31 +279,34 @@ protected function hasTargetedBinding(string $interface, string $target = null)
...
+            if (isset($reflectionCache[$class])) {
+                list($constructor, $parameters) = $reflectionCache[$class];
+            } else {
+                $reflectionClass = new ReflectionClass($class);
+                if (!$reflectionClass->isInstantiable()) {
+                    throw new IocException(
+                        sprintf(
+                            "%s is not instantiable%s",
+                            $class,
+                            $this->currentTarget === null ? "" : " (dependency of {$this->getCurrentTarget()})"
```
Comment (opulencephp) [inline src/Opulence/Ioc/Container.php]: Not sure this will work because we removed `getCurrentTarget()`.

**Comment by opulencephp** [comment]
At this point, the only thing keeping me from merging this is the static reflection cache.  It _might_ be a good idea to be static, but it also might make sense to be an instance property.  I haven't made up my mind which way to go, yet.

**Comment by alchimik** [comment]
The static reflection cache is only useful if we use multiple containers. But usually there is only one container. So, I think, it is better to make a  instance property.

**Pull request merged**